src/OrchardCore.Modules/Orchard.Setup/Views/Setup.cshtml

```diff
@@ -0,0 +1,218 @@
+﻿@using Orchard.Setup.ViewModels 
+@using System.Linq
+@using Orchard.Setup.Controllers
+@using Microsoft.AspNetCore.Builder
+@using Microsoft.AspNetCore.Html
+@using System.IO;
+@inject Microsoft.AspNetCore.Mvc.Localization.IViewLocalizer T
+@inject Microsoft.Extensions.Options.IOptions<IdentityOptions> identityOptions
+@model dynamic
+
+@functions{
+    void AddPrefixToModelState(string prefix) {
+        var originalModelState = ViewData.ModelState.ToList();
+        ViewData.ModelState.Clear();
+        foreach (var keyValuePair in originalModelState)
+        {
+            foreach (var error in keyValuePair.Value.Errors)
+            {
+                ViewData.ModelState.AddModelError(string.IsNullOrWhiteSpace(keyValuePair.Key) ? keyValuePair.Key : prefix+"."+keyValuePair.Key, error.ErrorMessage);
+            }
+        }
+    }
+}
+@{
+    var setupViewModel = Model.SetupViewModel as SetupViewModel;
+    AddPrefixToModelState(nameof(SetupViewModel));
```

Comment:
Instead of this hack do you agree if I change calls to AddModelError within Controller for adding `nameof(SetupViewModel)` as Prefix